### DISCUSSION TITLE
[FEATURE REQUEST] html mail through smtp functions.

### DISCUSSION THREAD

**Discussion by samroffe** [saltstack/salt#67432] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Currently, when we send an email using smtp module or salt state, email body doesn't support HTML formatting and can only be send  using Text format. With HTML support, we can easily represent data in a clear way to highlight statistics without needing to download any attachments. 

**Describe the solution you'd like**
To be able to pass the HTML code directly over the name parameter in smtp.send_message,  or as a path to a Jinja template, so that more complex data can be extracted using Jinja and put into the HTML code. 

E.g:

Send_Email:
  smtp.send_msg:
     - name:  /tmp/email.j2
     - profile: my-smtp-login
     - recipient: XXXXX@xxxx
     - subject: SaltStack
    
**Describe alternatives you've considered**
One solution as of now, would be sending the HTML file as an attachment.

**Additional context**

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://join.slack.com/t/saltstackcommunity/shared_invite/zt-3av8jjyf-oBQ2M0vhXOhJpNpRkPWBvg)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by samroffe** [comment]
Any update on this?

**Comment by priyank-R** [comment]
@s0undt3ch is it ok if I work on this one ?
